File: client/src/_models/userParams.ts
import { User } from "./user";

export class UserParams{
    pageNumber=1;
    pageSize=5;
    minAge=18;
    maxAge=99;
    gender!: string;
    orderBy: string ='lastActive';

    constructor(user:User|null) {
        this.gender=user!.gender == 'male'?'female':'male';
    }
}

File: API/Helpers/UserParams.cs
﻿namespace API.Helpers
{
    public class UserParams: PaginationParams
    {
        public string UserName { get; set; }
        public string Gender { get; set; }
        public int MaxAge { get; set; } = 100;
        public int MinAge { get; set; } = 18;
        public string OrderBy { get; set; } = "lastActive";
    }
}
